nexus/src/db/model.rs

```diff
@@ -1004,6 +1005,82 @@ impl SiloUser {
     }
 }
 
+impl_enum_type!(
+    #[derive(SqlType, Debug, QueryId)]
+    #[postgres(type_name = "provider_type", type_schema = "public")]
+    pub struct SiloIdentityProviderTypeEnum;
+
+    #[derive(Copy, Clone, Debug, AsExpression, FromSqlRow, Serialize, Deserialize, PartialEq)]
+    #[sql_type = "SiloIdentityProviderTypeEnum"]
+    pub enum SiloIdentityProviderType;
+
+    // Enum values
+    Local => b"local"
+    Saml => b"saml"
+    Ldap => b"ldap"
+);
+
+impl ToString for SiloIdentityProviderType {
+    fn to_string(&self) -> String {
+        match self {
+            SiloIdentityProviderType::Local => "local".to_string(),
+            SiloIdentityProviderType::Saml => "saml".to_string(),
+            SiloIdentityProviderType::Ldap => "ldap".to_string(),
+        }
+    }
+}
```

Comment:
sad to have to do this when the byte strings are right there — maybe we can add this to the macro

Reply:
done in 1e879180